latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+    WorkflowMetadataDefaults,
+)
+from flytekit.exceptions import scopes as exception_scopes
+from flytekit.models import interface as interface_models
+from flytekit.models import literals as literals_models
+from flytekit.models import types as type_models
+from snakemake.dag import DAG
+from snakemake.target_jobs import encode_target_jobs_cli_args
+
+from latch.types import LatchAuthor, LatchFile, LatchMetadata, LatchParameter
+
+SnakemakeInputVal = Union[snakemake.io._IOFile]
+
+
+def variable_name_for_file(file: snakemake.io.AnnotatedString):
+    return file.replace("/", "_").replace(".", "_").replace("-", "_")
+
+
+def variable_name_for_value(
+    val: SnakemakeInputVal,
+    params: Union[snakemake.io.InputFiles, snakemake.io.OutputFiles] = None,
+) -> str:
+
+    if params:
+        for name, v in params.items():
+            if val == v:
+                return name
+
+    return variable_name_for_file(val.file)
+
+
+def snakemake_dag_to_interface(
+    dag: DAG, docstring: Optional[Docstring] = None
+) -> Interface:
+
+    outputs: Dict[str, Type] = {}
+
+    for target in dag.targetjobs:
+        for x in target.input:
+            outputs[variable_name_for_value(x, target.input)] = LatchFile
+
+    inputs: Dict[str, Type] = {}
+    for job in dag.jobs:
+
+        dep_outputs = []
+        for dep, dep_files in dag.dependencies[job].items():
+            for o in dep.output:
+                if o in dep_files:
+                    dep_outputs.append(o)
+
+        for x in job.input:
+            if x not in dep_outputs:
+                inputs[variable_name_for_value(x, job.input)] = (
+                    LatchFile,
+                    None,
+                )
+
+    return Interface(inputs, outputs, docstring=docstring)
+
+
+def binding_data_from_python(
+    expected_literal_type: type_models.LiteralType,
+    t_value: typing.Any,
+    t_value_type: Optional[type] = None,
+) -> literals_models.BindingData:
+
+    if isinstance(t_value, Promise):
+        if not t_value.is_ready:
+            return literals_models.BindingData(promise=t_value.ref)
+
+
+def binding_from_python(
+    var_name: str,
+    expected_literal_type: type_models.LiteralType,
+    t_value: typing.Any,
+    t_value_type: type,
+) -> literals_models.Binding:
+    binding_data = binding_data_from_python(
+        expected_literal_type, t_value, t_value_type
+    )
+    return literals_models.Binding(var=var_name, binding=binding_data)
+
+
+def transform_type(x: type, description: str = None) -> interface_models.Variable:
+    return interface_models.Variable(
+        type=TypeEngine.to_literal_type(x), description=description
+    )
+
+
+def transform_types_in_variable_map(
+    variable_map: Dict[str, type],
+    descriptions: Dict[str, str] = {},
+) -> Dict[str, interface_models.Variable]:
+    res = OrderedDict()
+    if variable_map:
+        for k, v in variable_map.items():
+            res[k] = transform_type(v, descriptions.get(k, k))
+    return res
```

Comment:
```suggestion
    res = OrderedDict()
    if len(variable_map) == 0:
    	return res
    
    for k, v in variable_map.items():
        res[k] = transform_type(v, descriptions.get(k, k))

    return res
```

Reply:
1. I dont see how using the dict itself as the truthy value will ever produce undesired behavior
2. your refactor adds a line of code for no reason?